strr-platform-web/app/components/connect/DetailsHeader.vue

```diff
@@ -0,0 +1,108 @@
+<script setup lang="ts">
+const {
+  loading,
+  title,
+  subtitles,
+  details,
+  sideDetails,
+  bottomButtons
+} = storeToRefs(useConnectDetailsHeaderStore())
+</script>
+
+<template>
+  <div class="bg-white py-5" data-testid="connect-details-header">
+    <div class="app-inner-container">
+      <div v-if="loading" class="flex animate-pulse flex-col gap-2 sm:flex-row">
+        <div class="grow space-y-2">
+          <div class="h-9 w-[400px] rounded bg-gray-200" />
+          <div class="h-5 w-[250px] rounded bg-gray-200" />
+          <div class="h-5 w-[200px] rounded bg-gray-200" />
+          <div class="h-5 w-[150px] rounded bg-gray-200" />
+        </div>
+        <div class="space-y-2">
+          <div class="h-5 w-[300px] rounded bg-gray-200" />
+          <div class="h-5 w-[300px] rounded bg-gray-200" />
+          <div class="h-5 w-[300px] rounded bg-gray-200" />
+          <div class="h-5 w-[300px] rounded bg-gray-200" />
+        </div>
+      </div>
+      <div v-else class="flex flex-col gap-2 sm:flex-row">
+        <div class="grow space-y-4">
+          <div class="space-y-2">
+            <ConnectTypographyH1 v-if="title" :text="title" />
+            <div v-if="subtitles.length" class="flex divide-x *:border-gray-500 *:px-2 first:*:pl-0">
+              <p v-for="subtitle in subtitles" :key="subtitle" class="text-sm">
+                {{ subtitle }}
+              </p>
+            </div>
+          </div>
+          <div class="space-y-2">
+            <slot name="details">
+              <p v-if="details" class="text-sm">
+                {{ details }}
+              </p>
+            </slot>
+            <slot name="buttons">
+              <div v-if="bottomButtons.length" class="flex flex-wrap gap-2">
+                <UButton
+                  v-for="btn in bottomButtons"
+                  :key="btn.label"
+                  :label="btn.label"
+                  :icon="btn.icon"
+                  class="pl-0"
+                  color="primary"
+                  variant="link"
+                  @click="btn.action()"
+                />
+              </div>
+            </slot>
+          </div>
+        </div>
+        <dl class="mr-14 space-y-1 text-sm">
+          <template v-for="detail in sideDetails" :key="detail.label">
+            <div>
+              <UTooltip
```

Comment:
This is a Scott question and I know we aren't adding this now maybe, but how does the user know this field editable? Should we have a little edit icon beside the BN?

Reply:
Yeah I was also thinking this needs something like that. Not going to put much more into this though until Scott has a chance to look at it more